cores/esp8266/IPAddress.cpp

```diff
@@ -112,4 +112,51 @@ String IPAddress::toString() const
     return String(szRet);
 }
 
+/* ref: https://code.woboq.org/userspace/glibc/resolv/inet_pton.c.html#inet_pton4 */
+static bool inet_pton4(const char *start, const char *end) {
```

Comment:
How is this function different from IPAddress::fromString(), found [here](https://github.com/esp8266/Arduino/blob/f4c391032aff382bcd243bb038d69feb136a96b5/cores/esp8266/IPAddress.cpp#L43)? If they serve equivalent purpose, I'd rather not duplicate.

Reply:
@devyte 
So sorry for late reply.
So you say we should modify `bool IPAddress::fromString(const char *address)` function for adding validate support ? without duplicate

Reply:
@devyte 
My suggestion:
1.Adding a private validating method:
`static bool IPAddress::isValid(const char* arg, size_t len);`
2.Adding correspond public method:
`static bool IPAddress::isValid(const String& ip);`
3.Reuse method of 1 inside 2 and `bool IPAddress::fromString(const char *address)`

Reply:
@IMAN4K said: 
> Because current IPAddress public API doesn't support validating IPV4 as string

Are you saying that the following does not validate a string?
`bool isvalid = IPAddress().fromString("17604123874106239");`

Reply:
@devyte 
Yes that's correct, my mistake!
But we need to make an object for every validation.
It's better with a public static member, then every where in code:
`if (IPAddress::isValid(...)) ...`

Reply:
The object constructed is 4 bytes in size, essentially one int, which is a temporary instantiated on the stack and immediately destroyed, so that's not a problem (btw, destruction is trivial). 
What I suggest is a static convenience method like you propose, but something like this:
```
bool IPAddress::isValid(const char *str)
{
  return IPAddress().fromString(str);
}
bool IPAddress::isValid(String &str)
{
  return IPAddress().fromString(str);
}

```

Reply:
Ok.
I'll make changes very soon.